packages/amplify-cli/src/plugin-helpers/platform-health-check.ts

```diff
@@ -0,0 +1,190 @@
+    name: 'hosting',
+    type: 'category',
+    packageName: 'amplify-category-hosting',
+    packageVersion: '',
+  },
+  interactions: {
+    name: 'interactions',
+    type: 'category',
+    packageName: 'amplify-category-interactions',
+    packageVersion: '',
+  },
+  notifications: {
+    name: 'notifications',
+    type: 'category',
+    packageName: 'amplify-category-notifications',
+    packageVersion: '',
+  },
+  predictions: {
+    name: 'predictions',
+    type: 'category',
+    packageName: 'amplify-category-predictions',
+    packageVersion: '',
+  },
+  storage: {
+    name: 'storage',
+    type: 'category',
+    packageName: 'amplify-category-storage',
+    packageVersion: '',
+  },
+  xr: {
+    name: 'xr',
+    type: 'category',
+    packageName: 'amplify-category-xr',
+    packageVersion: '',
+  },
+  codegen: {
+    name: 'codegen',
+    type: 'util',
+    packageName: 'amplify-codegen',
+    packageVersion: '',
+  },
+  android: {
+    name: 'android',
+    type: 'frontend',
+    packageName: 'amplify-frontend-android',
+    packageVersion: '',
+  },
+  ios: {
+    name: 'ios',
+    type: 'frontend',
+    packageName: 'amplify-frontend-ios',
+    packageVersion: '',
+  },
+  javascript: {
+    name: 'javascript',
+    type: 'frontend',
+    packageName: 'amplify-frontend-javascript',
+    packageVersion: '',
+  },
+  mock: {
+    name: 'mock',
+    type: 'util',
+    packageName: 'amplify-util-mock',
+    packageVersion: '',
+  },
+};
+
+export async function checkPlatformHealth(pluginPlatform: PluginPlatform): Promise<boolean> {
+  const activePlugins = pluginPlatform.plugins;
+  setOfficialPackageVersions();
+  const missingOfficialPlugins: Array<PluginDescription> = [];
+  const mismatchedOfficialPlugins: Array<PluginDescription> = [];
+
+  Object.keys(officialPlugins).forEach((plugin: string) => {
+    const officialPluginDescription = officialPlugins[plugin];
+    if (activePlugins[officialPluginDescription.name]) {
+      let isPackageMatching = false;
+      activePlugins[officialPluginDescription.name].every((pluginInfo: PluginInfo) => {
+        if (isMatching(officialPluginDescription, pluginInfo)) {
+          isPackageMatching = true;
+          return false;
+        }
+        return true;
+      });
+
+      if (!isPackageMatching) {
+        mismatchedOfficialPlugins.push(officialPluginDescription);
+      }
+    } else {
+      missingOfficialPlugins.push(officialPluginDescription);
+    }
+  });
+
+  if (missingOfficialPlugins.length > 0) {
+    console.log(chalk.yellow('The following official plugins are missing or inactive:'));
+    missingOfficialPlugins.forEach((pluginDescription: PluginDescription) => {
+      console.log('    ' + pluginDescription.name);
+    });
+  }
+
+  if (mismatchedOfficialPlugins.length > 0) {
+    console.log(chalk.yellow('The following official plugins have incorrect versions:'));
+    mismatchedOfficialPlugins.forEach((pluginDescription: PluginDescription) => {
```

Comment:
If we've a version mismatch it would be a better DX if we print what version did we expected and what version was found by the plugin platform. That would help the troubleshooting.

Reply:
I know, but that will need to go through API review, in the original design doc, those diagnostic info will be in the plugin platform's logs, but that's not implemented yet.

Reply:
did we ppush the change for this?

Reply:
yes, it's in line 52 of platform-health-check.ts